File: 算法/数论/算数基本定理.py
def prime_factors(n):
    '''算数基本定理:一个整数n可以被拆分成素数的乘积
    参数n，返回值为将n拆分的数组
    即该数组的乘积等于n'''
    factors = []
    divisor = 2
    while divisor <= n:
        if n % divisor == 0:
            factors.append(divisor)
            n = n // divisor
        else:
            divisor += 1

    return factors#返回值为数组

File: 算法/数论/算数基本定理.cpp
#include <iostream>
using namespace std;
int* prime_factors(int x, int& size) {//算数基本定理参数int x,size作为返回数组的长度

    int* a = new int[x];  // 使用 new 动态分配数组
    int cnt = 0;

    for (int i = 2; i <= x / i; i++) {
        while (x % i == 0) {
            a[cnt++] = i;
            x /= i;
        }
    }

    if (x > 1) {
        a[cnt++] = x;
    }

    size = cnt;  // 将数组大小传递给调用者

    return a;  // 返回指向动态分配数组的指针
}
//在函数外可用size作为数组长度.

int main() {
    int n ;
    int size;
    cin>>n;

    // 调用函数并获取返回的数组指针和大小
    int* factors = prime_factors(n, size);

    // 打印返回的数组元素
    for (int i = 0; i < size; ++i) {
        std::cout << factors[i] << " ";
    }

    // 记得释放动态分配的内存
    delete[] factors;

    return 0;
}